### PR TITLE
Localist Importer

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- Implementation of the Localist importer functionality.

This PR goes along with: 
- https://github.com/SU-SWS/stanford_events/pull/87
- https://github.com/SU-SWS/stanford_fields/pull/21

# Review By (Date)
- 11/19

# Criticality
- 10

# Urgency
- high

# Review Tasks

## Setup tasks and/or behavior to test

1. `composer require 'su-sws/stanford_profile:dev-D8CORE-4521 as dev-8.x-2.x; composer update`
2. alternately, checkout this branch, and the associated branches in `stanford_fields` and `stanford_events`
3. `drush cim` or, alternately, `blt drupal:update`
4. Rebuild Cache and import config `drush cr `
5. Navigate to: `/admin/config/importers/events-importer`
6.  You should notice a new field under the original "Events to be Imported" field, which should be labeled "Localist URL"
7. Under `Departments/Groups`, choose "Stanford Web Services".
8. At the bottom of the page, click "Save & Import".
9. That will pull in the feed from Localist with the events under the SWS department.  Right now, all that is available on the service is testing data.
10. Once the importers have run, verify that you have six new events in your database.  You will also find it has pulled in the vocabulary terms for event audiences and created new ones where necessary.  
11. There are tests in the stanford_fields module for the field widget.  Ensure they're passing as well.

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? YES

### no front end changes.
- [ ] Design is approved by @ user? 
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: config/sync/migrate_plus.migration.stanford_localist_importer.yml [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      callable: strtotime
+      source: event_instance_end
+  su_event_date_time/duration:
+    -
+      plugin: stanford_events_datemath
+      operation: subtraction
+      source: event_instance_end
+      values:
+        - event_instance_start
+  su_event_date_time/timezone: America/Los_Angeles
```
Comment (pookmish) [inline config/sync/migrate_plus.migration.stanford_localist_importer.yml:326]: does this have to be a constant?

**Review by pookmish — COMMENTED** [review]
Nothing really stands out to me. There will need to be changes to this based on my feedback in the stanford_fields PR

File: config/sync/migrate_plus.migration.stanford_localist_importer.yml (hunk 1 above)
Comment (imonroe) [reply config/sync/migrate_plus.migration.stanford_localist_importer.yml]: fxed.

Comment (imonroe) [reply config/sync/migrate_plus.migration.stanford_localist_importer.yml]: It's now using the `constants` syntax.

**Review by pookmish — APPROVED** [review]

**Pull request merged**